joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1331,17 +1368,31 @@ void JointTrajectoryController::fill_partial_goal(
         // Assume hold position with 0 velocity and acceleration for missing joints
         if (!it.positions.empty())
         {
-          if (
-            has_position_command_interface_ &&
-            !std::isnan(joint_command_interface_[0][index].get().get_value()))
+          const auto position_command_value_op =
+            joint_command_interface_[0][index].get().get_optional();
+          if (!position_command_value_op.has_value())
+          {
+            RCLCPP_DEBUG(
+              get_node()->get_logger(),
+              "Unable to retrieve position command value of joint at index %zu", index);
+          }
+          if (has_position_command_interface_)
```

Comment:
```suggestion
          else if (has_position_command_interface_)
```
shouldn't this go in the else branch?